concent_api/common/decorators.py

```diff
@@ -47,3 +51,15 @@ def wrapper(*args: Any, **kwargs: Any) -> None:
                 logging_level=LoggingLevel.ERROR)
             raise
     return wrapper
+
+
+def non_nesting_atomic(using: str=None, savepoint: bool=True) -> Atomic:
+    if (
+        settings.DETECT_NESTED_TRANSACTIONS and
+        get_connection(using if using is not None else DEFAULT_DB_ALIAS).in_atomic_block
+    ):
+        raise ConcentPendingTransactionError
+    if callable(using):
```

Comment:
could you please elaborate what this `if` is in fact checking and what happens when it is `True`?

Reply:
As mentioned above, just like in `transaction.atomic`, you can use it as both decorator and context manager. In the first case you get a function as first parameter. I have updated the typing :+1: done.